pycqed/analysis_v2/Two_state_T1_analysis.py

```diff
@@ -0,0 +1,333 @@
+
+        self.proc_data_dict['fit_msg'] = text_msg
+
+    def prepare_plots(self):
+        val_names = self.raw_data_dict['value_names']
+
+        for i, val_name in enumerate(val_names):
+            self.plot_dicts['plot_populations_{}'.format(val_name)] = {
+                'plotfn': plot_populations,
+                'time': self.proc_data_dict['time'],
+                'P0': self.proc_data_dict['P0'][val_name],
+                'P1': self.proc_data_dict['P1'][val_name],
+                'P2': self.proc_data_dict['P2'][val_name],
+
+                'xlabel': 'Time',
+                'xunit': self.raw_data_dict['time units'],
+                'ylabel': val_name,
+                'yunit': self.proc_data_dict['value_units'][i],
+                'title': self.proc_data_dict['timestamp_string']
+                          + '\n' +
+                          self.proc_data_dict['measurementstring']
+                }
+
+            # define figure and axes here to have custom layout
+
+            self.plot_dicts['fit_res_P0'] = {
+                'plotfn': self.plot_fit,
+                # 'plot_init': True,
+                'ax_id': 'plot_populations_{}'.format(val_name),
+                'fit_res': self.fit_res['fit_res_P0'],
+                'setlabel': r'P($|g\rangle$) fit',
+                'do_legend': True,
+                'color': 'C0',
+            }
+
+            self.plot_dicts['fit_res_P1'] = {
+                'plotfn': self.plot_fit,
+                # 'plot_init': True,
+                'ax_id': 'plot_populations_{}'.format(val_name),
+                'fit_res': self.fit_res['fit_res_P1'],
+                'setlabel': r'P($|e\rangle$) fit',
+                'do_legend': True,
+                'color': 'C1',
+
+            }
+
+            self.plot_dicts['fit_res_P2'] = {
+                'plotfn': self.plot_fit,
+                'ax_id': 'plot_populations_{}'.format(val_name),
+                'fit_res': self.fit_res['fit_res_P2'],
+                'setlabel': r'P($|f\rangle$) fit',
+                'do_legend': True,
+                'color': 'C2',
+            }
+
+            self.plot_dicts['fit_msg'] = {
+                'plotfn': self.plot_text,
+                'text_string': self.proc_data_dict['fit_msg'],
+                'xpos': 0.1, 'ypos': .9,
+                'ax_id': 'plot_populations_{}'.format(val_name),
+                'horizontalalignment': 'left'}
+
+
+def plot_populations(time, P0, P1, P2, ax,
+                     xlabel='Time', xunit='s',
+                     ylabel='Population', yunit='',
+                     title='', **kw):
+    ax.plot(time, P0, c='C0', linestyle='',
+            label=r'P($|g\rangle$)', marker='v')
+    ax.plot(time, P1, c='C1', linestyle='',
+            label=r'P($|e\rangle$)', marker='^')
+    ax.plot(time, P2, c='C2', linestyle='',
+            label=r'P($|f\rangle$)', marker='d')
+
+    set_xlabel(ax, xlabel, xunit)
+    set_ylabel(ax, ylabel)
+    ax.grid(axis='y')
```

Comment:
please remove the grid here. In certain other contexts it has meaning but here it is only visual bloat.

Reply:
Okay.